Mathlib/Analysis/Calculus/SecondDerivativeTest.lean

```diff
@@ -0,0 +1,142 @@
+/-
+Copyright (c) 2024 Bjørn Kjos-Hanssen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bjørn Kjos-Hanssen, Jireh Loreaux, Floris van Doorn, Eric Wieser
+-/
+import Mathlib.Analysis.Calculus.FirstDerivativeTest
+import Mathlib.Order.Interval.Set.Basic
+
+/-!
+# The Second-Derivative Test
+
+We prove the Second-Derivative test from calculus using the First-Derivative test.
+Source: [Wikipedia](https://en.wikipedia.org/wiki/Derivative_test#Proof_of_the_second-derivative_test).
+
+## Main results
+
+* `isLocalMin_of_deriv_deriv_pos`: The second-derivative test, minimum version.
+
+## Tags
+
+derivative test, calculus
+-/
+
+open Set Filter Topology SignType
+
+section SecondDeriv
+
+variable {f : ℝ → ℝ} {x₀ : ℝ}
+
+
+lemma slope_pos_iff {𝕜} [LinearOrderedField 𝕜] {f : 𝕜 → 𝕜} {x₀ b : 𝕜} (hb : x₀ < b) :
+    0 < slope f x₀ b ↔ f x₀ < f b := by
+  simp [slope, hb]
+
+lemma slope_pos_iff_gt {𝕜} [LinearOrderedField 𝕜] {f : 𝕜 → 𝕜} {x₀ b : 𝕜} (hb : b < x₀) :
+    0 < slope f x₀ b ↔ f b < f x₀ := by
+  rw [slope_comm, slope_pos_iff hb]
+
+lemma pos_of_slope_pos {b : ℝ} (hb : x₀ < b) (hbf : 0 < slope f x₀ b)
+    (hf : f x₀ = 0) : 0 < f b := by
+  simp_all [slope, hf]
+
+lemma neg_of_slope_pos {b : ℝ} (hb : b < x₀) (hbf : 0 < slope f x₀ b)
+    (hf : f x₀ = 0) : f b < 0 := by
+  simp_all [slope, hf]
+  exact neg_of_mul_pos_right hbf <| le_of_lt <| inv_lt_zero.mpr <| by linarith
+
+lemma neg_of_slope_neg {b : ℝ} (hb : b < x₀) (hbf : 0 < slope f x₀ b)
```

Comment:
This currently states the same as the previous lemma. I think you can delete this unless you actually use it.